### PR TITLE
Split storage modules

### PR DESCRIPTION
**Pull request by swsnr**
* Move modules and filters to `reference.conf`.
* Document settings in `reference.conf`.
* Refer to `reference.conf` and Play configuration docs in README
* Use pureconfig to load all configuration into a single large case class to allow modules type-safe access to their configuration and remove all those `fromConfig` methods.
* Extract storage bindings from the instance module to switch between storage types by enabling/disabling the corresponding module.

## Changes to the configuration

- `broccoli.instances.parameters.defaultType` becomes  `broccoli.instances.parameters.default-type` to appease pureconfig which uses dashes to convert between camelCase words in case class field names.  I think that's configurable, but imho it's simpler to change the key name.
- `broccoli.instances.storage.couchdb.dbName` becomes `broccoli.instances.storage.couchdb.database`. For the same reason.
- `broccoli.instances.storage.fs.url` becomes `broccoli.instances.storage.fs.path`.
- `broccoli.instances.storage.type` is gone.  To switch between storage type selectively enable/disable the corresponding storage modules (see `reference.conf`).

### CODE REVIEW

File: server/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1,16 +1,67 @@
...
+    enabled += "de.frosner.broccoli.templates.TemplateModule"
+    enabled += "de.frosner.broccoli.signal.SignalModule"
+    enabled += "de.frosner.broccoli.instances.InstanceModule"
+
+    # Configurable modules for Broccoli
+    # Store instances in file system
+    enabled += "de.frosner.broccoli.instances.storage.filesystem.FileSystemStorageModule"
+    # To store instances in CouchDB add the following to application.conf
+    # disabled += "de.frosner.broccoli.instances.storage.filesystem.FileSystemStorageModule"
+    # enabled += "de.frosner.broccoli.instances.storage.couchdb.CouchDBStorageModule"
```
Comment (swsnr) [inline server/src/main/resources/reference.conf:21]: Important change:  That's how to switch to couchdb storage now.

File: server/src/main/resources/reference.conf [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
+    # The filesystem path to load templates from
+    path = "./templates"
+  }
+
+  # How to configure and save instances
   instances {
+
+    # How to store instances
     storage {
-      type = "fs"
```
Comment (swsnr) [inline server/src/main/resources/reference.conf:7]: This one is gone now! Storage type is controlled by modules now, see above.

**Comment by swsnr** [comment]
@FRosner Last issue should be fixed now, the build should pass.  Please review and merge :blush:

**Comment by swsnr** [comment]
@FRosner Ping

File: README.md [hunk 3]
```diff
@@ -66,29 +66,17 @@ To build the production Docker image in a local Docker daemon run `sbt docker:pu
...
 The following configuration properties are supported. Please refer to the [Wiki](https://github.com/FRosner/cluster-broccoli/wiki) for more information about the individual topics.
 
+See [Play configuration](https://www.playframework.com/documentation/2.5.x/Configuration) and [Broccoli's reference.conf](https://github.com/FRosner/cluster-broccoli/blob/master/server/src/main/resources/reference.conf) for available parameters and their meanings.
+
 ### Nomad and Consul
 
 | Property | Description | Default |
 | -------- | ----------- | ------- |
-| `broccoli.nomad.url` | Address of your nomad server | `http://localhost:4646` |
 | `broccoli.consul.url` | Address of your consul server | `http://localhost:8500` |
```
Comment (FRosner) [inline README.md:78]: Why did you leave these config parameters here? Can't we make them part of the `reference.conf` as well?

**Review by FRosner — APPROVED** [review]

**Comment by FRosner** [comment]
@lunaryorn what happens when you enable multiple instance storage modules?

**Comment by FRosner** [comment]
I [updated the migration guide](https://github.com/FRosner/cluster-broccoli/wiki/Upgrading-to-0.7.0/_compare/edf6dde80baff2ded78d9c69e780d59b2411e900...547fbfa937376d55a1a4afcce6374e8fefb6b1ee) on the wiki. Did I cover everything?

**Review by Gerrrr — APPROVED** [review]

**Comment by swsnr** [comment]
@FRosner Broccoli will throw an exception because bindings are ambiguous.

**Comment by swsnr** [comment]
@FRosner I think you have it all in the migration guide.  Can't see anything missing.

**Pull request merged**